src/OpenRasta/UriTemplate.cs

```diff
@@ -336,28 +343,43 @@ public bool IsEquivalentTo(UriTemplate other)
 
     public UriTemplateMatch Match(Uri baseAddress, Uri uri)
     {
-      if (baseAddress == null || uri == null)
+      var baseLeft = baseAddress.GetLeftPart(UriPartial.Authority);
+      var baseSegments = baseAddress.Segments;
+      return Match(baseAddress, baseLeft, baseSegments, uri);
+    }
+
+    public UriTemplateMatch Match(Uri baseAddress, string baseLeft,string[] baseSegments, Uri uri)
+    {
+      if (uri == null)
         return null;
-      if (baseAddress.GetLeftPart(UriPartial.Authority) != uri.GetLeftPart(UriPartial.Authority))
+      if (baseLeft != uri.GetLeftPart(UriPartial.Authority))
         return null;
 
-      var baseUriSegments = baseAddress.Segments.Select(RemoveTrailingSlash);
-      var candidateSegments = new List<string>(uri.Segments.Select(RemoveTrailingSlash));
+      var segments = uri.Segments;
+      var candidateSegments = new List<string>(segments.Length);
+      foreach (var segment in segments)
+      {
+        candidateSegments.Add(RemoveTrailingSlash(segment));
+      }
+
+      var cullCount = 0;
+      foreach (var baseUriSegment in baseSegments)
+        if (RemoveTrailingSlash(baseUriSegment) == candidateSegments[cullCount])

```

Comment:
Could the `RemoveTrailingSlash` call be moved into the `UriTemplateTable.Match()` method [here](https://github.com/ololabs/openrasta-core/pull/2/commits/76652db93a40792efbc3bd3ece71fa54ecfd05f6#diff-8d432c06c8f221dca9a57755cbc09104R61) so that we don't need to trim each segment for each possible `UriTemplate` instance?

Reply:
Additionally, could the `RemoveTrailingSlash` call be simplified to just `baseUriSegment.TrimEnd('/')`?